site/_posts/2020-05-06-using-java-to-manage-fusionauth.md

```diff
@@ -0,0 +1,743 @@
+
+If you go to [this GitHub page](https://github.com/FusionAuth/fusionauth-java-client/blob/master/src/main/java/io/fusionauth/client/FusionAuthClient.java), you can see the various methods alongside their definitions. For example, to create an application, we’ll use the `createApplication` method. If you check the method definition, you’ll find that it only requires two parameters: the ID of the application and the request object that contains all the information used for creating the application. 
+
+That’s a simple way of consuming the FusionAuth REST API. Isn't it?
+
+
+Here is the code for using the createApplication method to create an application in FusionAuth:
+
+```java
+package io.fusionauth.example;
+
+import com.inversoft.error.Errors;
+import io.fusionauth.client.FusionAuthClient;
+import io.fusionauth.domain.Application;
+import io.fusionauth.domain.api.ApplicationRequest;
+import io.fusionauth.domain.api.ApplicationResponse;
+import com.inversoft.rest.ClientResponse;
+
+public class CreateApplication {
+
+    public static void main(String[] args) {
+
+        ApplicationProperties.setupProperties();
+
+        // Initiating the client
+        FusionAuthClient client = new FusionAuthClient(ApplicationProperties.getApiKey(),
+                ApplicationProperties.getFusionAuthURL());
+
+        // Initiating the application and providing registration details
+        Application app = new Application();
+
+        app.name = "Todo";
+
+        // Creating the request object
+        ApplicationRequest request = new ApplicationRequest(app, null);
+
+        // Using the returned ClientResponse object
+        ClientResponse<ApplicationResponse, Errors> response = client.createApplication(null, request);
+
+        if (response.wasSuccessful()) {
+            System.out.println("Application creation successful");
+        } else {
+            // Handling errors
+            System.out.println(response.exception);
+        }
+    }
+
+}
+```
+As you can see from the code above, when initiating the FusionAuth client, we specified the API key and the web address that points to our running FusionAuth application. You will need to provide your own unique values for these parameters.
+
+Then, we created an `app` object from the `Application` class, and used the object to define the name of the application. We also passed this object as a parameter to the `ApplicationRequest` class. 
+
+Notice that the Java client returns a `ClientResponse` object that contains everything that happened while communicating with the FusionAuth server. 
+
+Also, we passed the value of the first parameter of the `createApplication` method as `null`. This will allow FusionAuth to generate for us a secure random UUID for the created application. 
```

Comment:
Do we need "for us" in the second sentence? If so, consider revising to:

This will allow FusionAuth to generate a secure random UUID we can use for the created application.

Reply:
nope, I'll remove.